indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -707,6 +708,31 @@ def _process(self):
         return jsonify_data(html=_render_registration_details(self.registration))
 
 
+class RHRegistrationAllowModification(RHManageRegistrationBase):
+    """Allow the participant to modify their registration and add a deadline."""
+
+    def _process(self):
+        event_end_dt = self.regform.event.end_dt
+        default_modification_end_dt = FormDefaults(modification_end_dt=event_end_dt)
+        form = RegistrationExceptionalModificationForm(regform=self.regform, obj=default_modification_end_dt)
+
+        if form.validate_on_submit():
+            if form.modification_end_dt.data:
+                self.registration.exceptional_modification_allowed_end_dt = form.modification_end_dt.data
+            else:
+                self.registration.exceptional_modification_allowed_end_dt = event_end_dt
+            return jsonify_data(html=_render_registration_details(self.registration), flash=False)
+        return jsonify_form(form, submit=_('Set'), disabled_until_change=False)
+
+
+class RHRegistrationRemoveAllowModification(RHManageRegistrationBase):
+    """Let a manager toggle allowance to modify a registration."""
```

Comment:
Maybe this could be reworded to better fit what this does?

```suggestion
class RHToggleRegistrationModification(RHManageRegistrationBase):
    """Enable managers to toggle permission for modifying registrations."""
```

Reply:
Thanks, this does make it clearer to understand.

Reply:
@Thanhphan1147 You need to apply this manually (your old class name `RHRegistrationRemoveAllowModification` is still in the blueprint) :P

Reply:
I reverted the class name to the old class name as it only removes the permission rather than toggle it, I also updated the docstring to match

Reply:
In that case, I would suggest:

```py
class RHDenyRegistrationModification(RHManageRegistrationBase)
```